lib/pure/browsers.nim

```diff
@@ -13,14 +13,46 @@
 ## Unstable API.
 
 import std/private/since
-
+from os import findExe
 import strutils
 
 when defined(windows):
   import winlean
 else:
   import os, osproc
 
+since (1, 3):
+  type WebBrowsers* = enum          ## Web Browsers
```

Comment:
I think that such hard coded list of browsers is not very future proof, so not very stdlib-apt. Wouldn't xdg-open or alike be more appropriate?

Reply:
`openDefaultBrowser()` already uses `xdg-open`.
This is to open with browsers that are not the default one.

Reply:
What Type do you suggest for Enumeration better than `enum` ?.
It can be a big `seq[string]` as default argument, but I dont think thats better neither.

This is not to have all browsers, just the most *well known* popular ones,
because browsers are open source you can theoretically create your own, 
so is impossible to known them all. 

[Looking at other langs it does not looks like a bad implementation in general.](https://github.com/python/cpython/tree/3.8/Lib/webbrowser.py)

Reply:
IMO, I was thinking of an stdlib as something "solid". Here, it's relying on a lot of 3rd party projects. New open source projects get added and removed, and that would affect the quality of this stdlib. 

Also, why does this need to be an enum? Why not allow user to do `openBrowser("https://nim-lang.org", "firefox", "-new-tab")`?

The list of browsers looks over-specified. If a user is using a non-popular browser, they cannot use this proc. But if they have that non-popular browser named "foo" in their PATH, they can just do `openBrowser("https://nim-lang.org", "foo")`.